angular-workspace/projects/ni/nimble-angular/table-column/date-text/tests/nimble-table-column-date-text.directive.spec.ts

```diff
@@ -0,0 +1,1671 @@
+                    </nimble-table-column-date-text>
+                </nimble-table>
+            `
+        })
+        class TestHostComponent {
+            @ViewChild('column', { read: NimbleTableColumnDateTextDirective }) public directive: NimbleTableColumnDateTextDirective;
+            @ViewChild('column', { read: ElementRef }) public elementRef: ElementRef<TableColumnDateText>;
+        }
+
+        let fixture: ComponentFixture<TestHostComponent>;
+        let directive: NimbleTableColumnDateTextDirective;
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [TestHostComponent],
+                imports: [NimbleTableColumnDateTextModule, NimbleTableModule]
+            });
+
+            fixture = TestBed.createComponent(TestHostComponent);
+            fixture.detectChanges();
+            directive = fixture.componentInstance.directive;
+        });
+
+        it('has valid configuration by default', () => {
+            expect(directive.checkValidity()).toBeTrue();
+            expect(directive.validity.invalidCustomOptionsCombination).toBeFalse();
+        });
+
+        it('has invalid configuration with both customDateStyle and customMonth set', () => {
+            directive.format = 'custom';
+            directive.customDateStyle = 'full';
+            directive.customMonth = 'long';
+            expect(directive.checkValidity()).toBeFalse();
+            expect(directive.validity.invalidCustomOptionsCombination).toBeTrue();
+        });
+    });
+
+    describe('with template string values', () => {
+        @Component({
+            template: `
+                <nimble-table>
+                    <nimble-table-column-date-text
+                        #column
+                        column-id="my-column"
+                        field-name="field1"
+                        action-menu-slot="my-slot"
+                        action-menu-label="my menu"
+                        column-hidden="true"
+                        fractional-width="2"
+                        min-pixel-width="40"
+                        sort-direction="${TableColumnSortDirection.ascending}"
+                        sort-index="0"
+                        group-index="0"
+                        grouping-disabled
+                        format="custom"
+                        custom-locale-matcher="lookup"
+                        custom-weekday="narrow"
+                        custom-era="long"
+                        custom-year="numeric"
+                        custom-month="short"
+                        custom-day="2-digit"
+                        custom-hour="numeric"
+                        custom-minute="2-digit"
+                        custom-second="numeric"
+                        custom-time-zone-name="shortOffset"
+                        custom-format-matcher="basic"
+                        custom-hour12="24hour"
```

Comment:
Why is `custom-hour12` being set to `24hour`? That isn't a valid value, right? And the test is asserting this gets coerced to `false`. I'm not against having a few extra tests for the conversion of this property -- in fact that would probably be a good thing. But, I think this test should use valid values.

Right now the property is defined as
```
@Input('custom-hour12') public set customHour12(value: boolean | string | undefined)
```
Should that be stricter on which strings are allowed?
```
@Input('custom-hour12') public set customHour12(value: boolean | 'true' | 'false' | undefined)
```

Reply:
> Why is `custom-hour12` being set to `24hour`?

That was a mistake.

> Should that be stricter on which strings are allowed?

Yeah, that makes sense to me.